src/Gui/SelectionObject.h

```diff
@@ -72,8 +72,10 @@ class GuiExport SelectionObject : public Base::BaseClass
 
     /// returns the selected DocumentObject or NULL if the object is already deleted
     const App::DocumentObject *getObject() const;
-    /// returns the selected DocumentObject or NULL if the object is already deleted
+    /// returns the selected DocumentObject as specified type or NULL if the object is already deleted
     App::DocumentObject *getObject();
+    template <class T>
+    T* getObject() { return Base::freecad_dynamic_cast<T>(getObject()); }

```

Comment:
For the sake of consistency there should also be a:
`template <class T> const T* getObject() const { return Base::freecad_dynamic_cast<T>(getObject()); }`

Reply:
I'll do that in https://github.com/FreeCAD/FreeCAD/pull/17528